cart/views.py

```diff
@@ -1,13 +1,32 @@
 from django.views.generic import TemplateView
+from django.views.generic.list import ListView
+from django.views.generic.detail import DetailView
+from cart.models import Product, Cart
+from django.template import loader
+from django.shortcuts import render
+from django.http import HttpResponseRedirect, HttpRequest
+from .forms import TickForm
+class ProductDetailView(DetailView):
+    model = Product
 
 
-class ProductListView(TemplateView):
-    template_name = 'cart/product-list.html'
-
-
-class ProductDetailView(TemplateView):
     template_name = 'cart/product-detail.html'
+    
+	
 
+    def get_context_data(self, **kwargs):
+       context = super().get_context_data(**kwargs)
+        
+       return context
```

Comment:
```
from .forms import CartForm


class ProductDetailView(DetailView):
    model = Product
    template_name = 'cart/product-detail.html'

    def get_context_data(self, **kwargs):
        context = super().get_context_data(**kwargs)
        context['form'] = CartForm(initial={'product': self.get_object()})
        return context

    def post(self, request, *args, **kwargs):
        form = CartForm(self.request.POST)
        self.object = self.get_object()
        if form.is_valid():
            form.save()
            return HttpResponseRedirect('cart:cart')
        context = self.get_context_data(**kwargs)
        context['form'] = form
        return self.render_to_response(context)
```

Reply:
This is how a form is passed to the template and evaluated in a class-based view. Ping me if you have any doubts regarding this.